include/aws/common/byte_buf.h

```diff
@@ -882,6 +882,46 @@ AWS_COMMON_API bool aws_isxdigit(uint8_t ch);
  */
 AWS_COMMON_API bool aws_isspace(uint8_t ch);
 
+/**
+ * Read entire cursor as ASCII/UTF-8 unsigned base-10 number.
+ * Stricter than strtoull(), which allows whitespace and inputs that start with "0x"
+ *
+ * Examples:
+ * "0" -> 0
+ * "123" -> 123
+ * "00004" -> 4 // leading zeros ok
+ *
+ * Rejects things like:
+ * "-1" // negative numbers not allowed
+ * "1,000" // only characters 0-9 allowed
+ * "" // blank string not allowed
+ * " 0 " // whitespace not allowed
+ * "0x0" // hex not allowed
+ * "FF" // hex not allowed
+ * "999999999999999999999999999999999999999999" // larger than max u64
+ */
+AWS_COMMON_API
+int aws_byte_cursor_parse_uint64(struct aws_byte_cursor cursor, uint64_t *dst);
```

Comment:
We need the name to be clear

We don't want this to be confused with functions like like `aws_byte_cursor_read_u8()` or `aws_byte_cursor_read_u64()` that reads the bytes as binary. Have something like "str" or "utf8" in the name to be clear we're operation on a byte-cursor that contains a string

also, if you're going to have "uint64" in the name, use "u64" instead to be consistent with other functions in this file.

maybe: `aws_byte_cursor_utf8_parse_u64()`? `aws_byte_cursor_parse_str_to_u64()`?

Reply:
Renamed to `aws_byte_cursor_utf8_parse_u64` , thanks!